### ISSUE TITLE
Spacy is not working with the version 8.10 of click.

### ISSUE THREAD

**Issue by zwitterion** [pallets/click#2226]
Environment:

- Python version: 3.9 and 3.8
- Click version: 8.10

**Comment by philgooch** [comment]
See also https://github.com/explosion/spaCy/issues/10564

**Comment by davidism** [comment]
When reporting an issue, be sure to include a minimal reproducible example, and a description of the problem, as it relates to the library.

The referenced function was previously deprecated, and is now removed. https://werkzeug.palletsprojects.com/en/8.1.x/changes/#version-8-1-0

8.0.0:

> `click.get_terminal_size()` is deprecated and will be removed in 8.1. Use [`shutil.get_terminal_size()`](https://docs.python.org/3/library/shutil.html#shutil.get_terminal_size) instead. [#1736](https://github.com/pallets/click/issues/1736)

8.1.0:

> Remove previously deprecated code. [#2130](https://github.com/pallets/click/pull/2130)
> * get_terminal_size is removed, use shutil.get_terminal_size instead.

You should use a tool like [pip-tools](https://pypi.org/project/pip-tools/) to pin your dependencies and control when you get updates. Be sure to run tests with deprecation warnings treated as errors so that you get notified of these types of changes early.

**Comment by henryiii** [comment]
Typer has not addressed this yet: https://github.com/tiangolo/typer/pull/375 (and duplicates) are not in yet, much yet released.

Would something like this be acceptable to provide a transition period?

```python
def get_terminal_size(*args, **kwargs):
    msg = "Deprecated, please use shutil.get_teriminal_size instead"
    warnings.warn(msg, FutureWarning, stacklevel=2)
    from shutil import get_terminal_size
    return get_terminal_size(*args, **kwargs)
```

All typer apps are broken at the moment, and this is not an underscore import, like the one that broke black. Just curious. Actually, guessing from above, something like this was probably there, but probably was using a 'lesser' warning level.

FutureWarning is the correct warning to provide for a removal that is about to happen. According to PEP 0565, there are [three levels of deprecation warnings](https://peps.python.org/pep-0565/#additional-use-case-for-futurewarning), with FutureWarning being the highest level. Ideally, you step through two or three levels, but the final level should be FutureWarning, since it shows up for everyone.

Also, stating people should lock dependencies does't help if it's a library that's broken (Typer, in this case). If you start a brand new project, pip install typer is broken. Pinning via a typer app via pre-commit is broken, etc. Libraries can't lock their dependencies[^1], only apps can.

[^1]: Libraries can lock them in their own CI. But not for users.

**Comment by henryiii** [comment]
Usages like this are breaking:

```yaml
- repo: https://github.com/hadialqattan/pycln
  rev: "v1.2.5"
  hooks:
  - id: pycln
    args: [--all]
```

https://scikit-hep.org/developer/style#pycln


There's no way to use pip-tools with something like that. I personally already use locking (via PDM mostly) for applications where it matters to me.

I would really recommend using at least a single release of FutureWarning before removing something without an underscore - obviously, what was done wan't enough for typer, which is a major dependent library. While I like warnings as errors and recommend it (https://scikit-hep.org/developer/pytest#configuring-pytest), it's not that common - many users don't know about it or find it annoying to enable[^1]. Also, due to the [despicable practice](https://iscinumpy.dev/post/bound-version-constraints/) of capping dependencies (heavily pushed by Poetry), many users are just now removing/updating `^7` caps to `^8`, causing them to completely miss the `8.0` release and jump to 8.1. FYI, PyTest also removes deprecated items in the `.1` releases.

I've already spent two days updating 30+ packages pre-commit's to use black 22.3.0, so the 4-5 packages that needed the extra pin added to pycln was a bit annoying, but it is not an issue for me, this was just suggesting something that would be nice for the other users of click via Typer. And the fact our developer guidelines are suggesting something that is broken is also bothersome, thought our cookiecutter already has the current workaround pin. And it was technically these libraries fault for not having warnings enabled, though I like to err on "being nice" over being "correct".

[^1]: See, for example, https://github.com/scikit-hep/pyhf/pull/1773#pullrequestreview-878643489

**Comment by henryiii** [comment]
FYI, I mentioned this to Sebastián and he replied "I'm on it!". So this hopefully will be fixed there soon. :)

**Comment by henryiii** [comment]
Typer 0.4.1 is out. :)

**Comment by tiangolo** [comment]
Hello! I just released Typer `0.4.1` that should handle this. :rocket: :nerd_face:
